src/main/java/com/hivemq/configuration/reader/RestrictionConfigurator.java

```diff
@@ -31,12 +36,56 @@ public RestrictionConfigurator(@NotNull final RestrictionsConfigurationService r
+        restrictionsConfigurationService.setMaxConnections(validateMaxConnections(restrictionsEntity.getMaxConnections()));
+        restrictionsConfigurationService.setMaxClientIdLength(validateMaxClientIdLength(restrictionsEntity.getMaxClientIdLength()));
+        restrictionsConfigurationService.setNoConnectIdleTimeout(validateNoConnectIdleTimeout(restrictionsEntity.getNoConnectIdleTimeout()));
+        restrictionsConfigurationService.setIncomingLimit(validateIncomingLimit(restrictionsEntity.getIncomingBandwidthThrottling()));
+        restrictionsConfigurationService.setMaxTopicLength(validateMaxTopicLength(restrictionsEntity.getMaxTopicLength()));
+    }
+
+    private long validateMaxConnections(final long maxConnections) {
+        if (maxConnections == UNLIMITED_CONNECTIONS) {
+            return maxConnections;
+        }
+        if (maxConnections < MAX_CONNECTIONS_MINIMUM) {
+            log.warn("The configured max-connections ({}) must be greater than 0. The value was set to {} (unlimited) instead.", maxConnections, MAX_CONNECTIONS_DEFAULT);
+            return MAX_CONNECTIONS_DEFAULT;
+        }
+        return maxConnections;
+    }
+
+    private int validateMaxClientIdLength(final int maxClientIdLength) {
+        if (maxClientIdLength < MAX_CLIENT_ID_LENGTH_MINIMUM || maxClientIdLength > MAX_CLIENT_ID_LENGTH_MAXIMUM) {
+            log.warn("The configured max-clientid-length ({}) must be in the range {} - {}. The value was set to {} instead.", MAX_CLIENT_ID_LENGTH_MINIMUM, MAX_CLIENT_ID_LENGTH_MAXIMUM, maxClientIdLength, MAX_CLIENT_ID_LENGTH_DEFAULT);
+            return MAX_CLIENT_ID_LENGTH_DEFAULT;
+        }
+        return maxClientIdLength;
+    }
+
+    private int validateMaxTopicLength(final int maxTopicLength) {
+        if (maxTopicLength < MAX_TOPIC_LENGTH_MINIMUM || maxTopicLength > MAX_TOPIC_LENGTH_MAXIMUM) {
+            log.warn("The configured max-topic-length ({}) must be in the range {} - {}. The value was set to {} instead.", MAX_TOPIC_LENGTH_MINIMUM, MAX_TOPIC_LENGTH_MINIMUM, maxTopicLength, MAX_TOPIC_LENGTH_DEFAULT);
+            return MAX_TOPIC_LENGTH_DEFAULT;
+        }
+        return maxTopicLength;
+    }
+
+    private long validateNoConnectIdleTimeout(final long noConnectIdleTimeout) {
+        if (noConnectIdleTimeout < NO_CONNECT_IDLE_TIMEOUT_MINIMUM) {
+            log.warn("The configured no-connect-idle-timeout ({}) must be greater than 0. The value was set to {} instead.", noConnectIdleTimeout, NO_CONNECT_IDLE_TIMEOUT_DEFAULT);
+            return NO_CONNECT_IDLE_TIMEOUT_DEFAULT;
+        }
+        return noConnectIdleTimeout;
     }
 
+    private long validateIncomingLimit(final long incomingLimit) {
+        if (incomingLimit < INCOMING_BANDWIDTH_THROTTLING_MINIMUM) {
+            log.warn("The configured incoming-bandwidth-throttling ({} bytes/second) must not be negative. The value was set to {} bytes/second instead.", incomingLimit, INCOMING_BANDWIDTH_THROTTLING_DEFAULT);
```

Comment:
Why does this message contain units and the others do not?

Reply:
no-connect-idle-timout can have 'ms' as a unit.